ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/util/ScalablePushUtilTest.java

```diff
@@ -252,7 +390,60 @@ public void isScalablePushQuery_false_wrongUpstreamQueries_Two() {
 
     // Then:
     assertThat(ScalablePushUtil.isScalablePushQuery(query, ksqlEngine, ksqlConfig,
-        ImmutableMap.of(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest")),
+            ImmutableMap.of(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest")),
         equalTo(false));
   }
-}
+
+  private void expectIsSQP() {
+    when(ksqlConfig.getBoolean(KsqlConfig.KSQL_QUERY_PUSH_V2_ENABLED)).thenReturn(true);
+    when(query.getGroupBy()).thenReturn(Optional.empty());
+    when(query.getWindow()).thenReturn(Optional.empty());
+    when(query.getHaving()).thenReturn(Optional.empty());
+    when(query.getPartitionBy()).thenReturn(Optional.empty());
+    when(query.getRefinement())
+        .thenReturn(Optional.of(RefinementInfo.of(OutputRefinement.CHANGES)));
+    when(query.getFrom())
+        .thenReturn(new AliasedRelation(new Table(SourceName.of("Foo")), SourceName.of("blah")));
+    when(ksqlEngine.getQueriesWithSink(SourceName.of("Foo"))).thenReturn(
+        ImmutableSet.of(new QueryId("a")));
+  }
+
+  private void givenScalablePushQuery() {
```

Comment:
How is this different from `expectIsSQP()`? Let's de-dup (especially since the names as-is are rather confusing).

Reply:
This presents an interesting issue- if we enable the feature flag, we need to mock the static `ColumnExtractor` using a try block, and this needs to be done in every test (unless there is a better way to mock a classes' static behavior that I'm not aware of, which is a strong possibility).

One workaround is disabling the flag for all tests which aren't testing the last condition, so true is immediately returned from `containsDisallowedColumns()` and it never tries to use the `ColumnExtractor`. Seems a bit hacky though, WDYT?

Reply:
How about enabling the feature flag just for the relevant tests? The default (for the rest of the tests) can be to leave it off.

Reply:
Ah I see you've already done this in your recent commit. Is there still an open question here?

Reply:
Nope, just wanted to check if that was a reasonable course of action.